storm-core/src/jvm/backtype/storm/task/TopologyContext.java

```diff
@@ -205,32 +217,65 @@ public Object getTaskData(String name) {
     public void setExecutorData(String name, Object data) {
         _executorData.put(name, data);
     }
-    
+
     public Object getExecutorData(String name) {
         return _executorData.get(name);
-    }    
-    
+    }
+
     public void addTaskHook(ITaskHook hook) {
         hook.prepare(_stormConf, this);
         _hooks.add(hook);
     }
-    
+
     public Collection<ITaskHook> getHooks() {
         return _hooks;
     }
+
+    public Object groupingToJSONableObject(Grouping grouping) {
+    	if (grouping.is_set_fields()) {
+    		return grouping.get_fields();
+    	} else {
+    		return grouping.getSetField().toString();
+    	}
```

Comment:
I was thinking it would look kind of like.
```{"type":"SHUFFLE"}```
instead of having the fields be null.  type would be the only required field, but I am open to other ideas.